CMSgov/onemac
# OY24856 - rai response attachments

Story: https://qmacbis.atlassian.net/browse/OY2-24856
Endpoint: See github-actions bot comment

### Details

Allow same attachment types from original submission as optional attachments for subsequent submissions on RAI Responses.

### Changes

- Added waiverAuthority as property on waiver rai response forms so that it can be passed into submit
- Added passing of waiverAuthority and waiverParentType in location state in rai response action
- Added method in waiver rai response config in common to determine conifg based on authority and type
- Updated WaiverRaiForm to compile formConfig based on waiver type and waiver authority using common method
- Updated Medicaid, and Appk configs to override optional attachments using parent config attachments
- Updated submitRaiResponse to use config based on waiver type and waiver authority using common method

### Implementation Notes

- I used a slightly different pattern to determine proper form config for waivers such that we didnt need an individual form and submit method per waiver type and waiver authority

### Test Plan

1. Login as state submitter
2. Find a package in RAI Response status (Medicaid Spa, Initial Waiver (B/B4), Renewal Waiver (B/B4), Amendment Waiver (B/B4), Appendix K)
3. Select Respond to RAI action
4. Verify attachments have the one required attachment = RAI Respone and optional attachments = original submission attachment types
5. Submit and verify detail page for the package has the RAI Response with the attachments
